server-common/src/test/java/org/apache/kafka/server/immutable/DelegationChecker.java

```diff
@@ -139,6 +163,16 @@ public void doFunctionDelegationCheck() {
         }
     }
 
+    public void doUnsupportedFunctionDelegrationCheck() {
```

Comment:
`s/Delegration/Delegation/` (and same above in `doUnsupportedVoidFunctionDelegrationCheck()`).

I think what is going on with these methods you added is that you are configuring the mock to invoke the real `TreePSet` method and then confirming that the invocation of the real method throws `UnsupportedOperationException`.  The methods that (for example) `PCollectionsImmutableMapTest.testDelegationOfPut()` were previously invoking were not invoking the underlying `HashPMap` method but were instead configuring the mock to flag the fact that the delegation occurred as expected (by setting `persistentCollectionMethodInvokedCorrectly = true`).

I think on the one hand we don't have to test that the real method throws the exception -- that's up to the library itself to test in its test suite.  All we have to test is that we delegate correctly.  This is why I implemented it this way.  However, I think it is useful to explicitly call out the fact that these methods will throw that exception.  So I'm fine with this change in general.  We just have to correct the `Delegration` misspelling.  Thanks for the improvement.

What do you think about adjusting the test method names where we expect the method to throw the exception?  So a test method might be `testDelegationOfUnsupportedOperationAdd()`?  I think this might be a good change.

Reply:
yup, I thought it made sense to test it explicitly.

Updated the functions.